### PR TITLE
Added nether biome whitelist/blacklist config option

### PR DESCRIPTION
**Pull request by caelwarner**
- Allows players to create a whitelist or blacklist of biomes that spawn in the nether
- Added WorldGenerationScreen GUI

### CODE REVIEW

**Comment by Dariensg** [comment]
LGTM, we need to make sure that all the other lang files are also updated to reflect this. I can take care of that in a matter of minutes once this is merged if you would like.

**Comment by caelwarner** [comment]
I say lets wait until we have actually finalized everything and are ready to release the beta. Unless it's easier to just do it now.

File: src/main/java/com/nekomaster1000/infernalexp/world/dimension/ModNetherBiomeCollector.java [hunk 1]
```diff
@@ -16,13 +18,17 @@
...
+    public static List<String> whitelistOrBlacklist = Arrays.asList(((String) InfernalExpansionConfig.WorldGeneration.BIOMES_WHITELIST_OR_BLACKLIST.get()).replace(" ", "").split(","));
+    public static boolean whitelistEnabled = (Boolean) InfernalExpansionConfig.WorldGeneration.BIOMES_WHITELIST_ENABLED.get();
 
     public static List<RegistryKey<Biome>> netherBiomeCollection(Registry<Biome> biomeRegistry) {
 
         for (Map.Entry<RegistryKey<Biome>, Biome> entry : biomeRegistry.getEntries()) {
             if (entry.getValue().getCategory() == Biome.Category.NETHER && !entry.getKey().getLocation().getNamespace().equals("ultra_amplified_dimension")) {
                 if (!netherBiomeList.contains(entry.getKey())) {
-                    netherBiomeList.add(entry.getKey());
+                    if ((whitelistEnabled && whitelistOrBlacklist.contains(entry.getKey().getLocation().toString())) || (!whitelistEnabled && !whitelistOrBlacklist.contains(entry.getKey().getLocation().toString())) || whitelistOrBlacklist.isEmpty()) {
```
Comment (SwanX1) [inline src/main/java/com/nekomaster1000/infernalexp/world/dimension/ModNetherBiomeCollector.java]: would've preferred some line breaks/indentation here

File: src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java [hunk 2]
```diff
@@ -282,17 +286,32 @@
...
-        
+
+        builder.pop();
+
+        // World Generation
+        builder.push("World Generation");
+
+        biomesWhitelistEnabled = builder
+            .comment("Should the biome list below act as a whitelist (True/On), or a blacklist (False/Off). CHANGING THIS REQUIRES A GAME RESTART.")
+            .translation(InfernalExpansion.MOD_ID + ".config.tooltip.biomesWhitelistEnabled")
```
Comment (SwanX1) [inline src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java]:
```suggestion
            .comment("Determines if the biome list below should act as a whitelist, or, when disabled, a blacklist.")
            .comment("CHANGING THIS REQUIRES A GAME RESTART.")
            .translation(InfernalExpansion.MOD_ID + ".config.tooltip.biomesWhitelistEnabled")
```

File: src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java [hunk 3]
```diff
@@ -50,7 +50,11 @@
     final ForgeConfigSpec.ConfigValue<String> glowsilkMothBiomes;
     final ForgeConfigSpec.ConfigValue<String> blindsightBiomes;
     final ForgeConfigSpec.ConfigValue<String> blackstoneDwarfBiomes;
-    
+
+    // World Generation
+    final ForgeConfigSpec.BooleanValue biomesWhitelistEnabled;
```
Comment (SwanX1) [inline src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java]: `biomeListIsWhitelist` would be more appropriate and self-explanatory.

File: src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java [hunk 4]
```diff
@@ -50,7 +50,11 @@
...
+    final ForgeConfigSpec.ConfigValue<String> biomesWhitelistOrBlacklist;
```
Comment (SwanX1) [inline src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java]: in `ForgeConfigSpec.ConfigValue<T>`, `T` can be `List<String>`

Comment (SwanX1) [reply src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java]: If `biomesWhitelistEnabled` is renamed, rename this to `biomesList`

**Review by SwanX1 — CHANGES_REQUESTED** [review]
minor changes

File: src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java (hunk 4 above)
Comment (caelwarner) [reply src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java]: Does making it `List<String>` parse it automatically?

Comment (SwanX1) [reply src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java]: yes

Comment (caelwarner) [reply src/main/java/com/nekomaster1000/infernalexp/config/CommonConfig.java]: honestly, keeping it as a string is just easier for me. If I change it to be a list then I have to write code to parse it for the GUI

**Review by SwanX1 — APPROVED** [review]

**Comment by SwanX1** [comment]
Gonna merge this one

**Pull request merged**